planex/cmd/pin.py

```diff
@@ -30,91 +40,125 @@ def spec_and_lnk(repo_path, package_name):
             "Spec file for {} not present in {}/SPECS".format(
                 package_name, repo_path))
 
-    spec = Spec(specname)
-
     linkname = "%s.lnk" % partial_file_path
     link = Link(linkname) if os.path.isfile(linkname) else None
+    spec = planex.spec.load(specname, link=link, defines=RPM_DEFINES)
 
-    return spec, link
+    return spec
 
 
-def repository_of(spec_or_link):
-    """Return the Repository of the provided Spec source url or Link url.
-       None if spec_or_link is None"""
-    if isinstance(spec_or_link, Spec):
-        return Repository(spec_or_link.sources()[0][1])
-    if isinstance(spec_or_link, Link):
-        return Repository(spec_or_link.url)
-    if spec_or_link is None:
-        return None
-    else:
-        sys.exit(
-            "repository_of: got unexpected object {}".format(
-                spec_or_link))
+def repo_or_path(arg):
+    """
+    Heuristic. Parse URL:commitish into (URL, commitish) and anything else into
```

Comment:
This comment and the `argparse` help text state the separator is ":" but the code uses "#".

Reply:
Well spotted, thanks!